Mathlib/RingTheory/PowerSeries/WellKnown.lean

```diff
@@ -64,6 +65,87 @@ theorem map_invUnitsSub (f : R →+* S) (u : Rˣ) :
+  | succ d hd =>
+      rw [pow_add, pow_one, mul_comm, show Nat.succ d = d + 1 by rfl, ext_iff]; exact fun n => by
+        rw [hd, coeff_mul]; simp only [coeff_mk, one_mul]; rw [Nat.succ_add, Nat.choose_succ_succ,
+        ← Finset.sum_antidiagonal_choose_add]; simp only [Pi.one_apply, one_mul, Nat.cast_sum]
+
+/--
+The power series `mk fun n => Nat.choose (d + n) d`, whose multiplicative inverse is
+`(1 - X) ^ (d + 1)`.
+-/
+noncomputable def invOneSubPow : S⟦X⟧ˣ where
+  val := mk fun n => Nat.choose (d + n) d
+  inv := (1 - X) ^ (d + 1)
+  val_inv := by
+    rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow,
+      show (mk 1) * (1 - X : S⟦X⟧) = 1 by
+      rw [mul_comm, ext_iff]; exact fun n => by
+        by_cases hn : n = 0
+        · subst hn; simp only [coeff_zero_eq_constantCoeff, map_mul, map_sub, map_one,
+          constantCoeff_X, sub_zero, one_mul, coeff_one, ↓reduceIte]; rfl
+        · simp only [one_mul, map_sub, coeff_mk, coeff_one, hn, ↓reduceIte];
+          rw [show n = n - 1 + 1 by exact (Nat.succ_pred hn).symm, sub_mul];
+          simp only [one_mul, map_sub, coeff_mk, Pi.one_apply, coeff_succ_X_mul, sub_self]]
+    exact one_pow (d + 1)
+  inv_val := by
+    rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow,
+      show (1 - X : S⟦X⟧) * (mk 1) = 1 by
+      rw [ext_iff]; exact fun n => by
+        by_cases hn : n = 0
+        · subst hn; simp only [coeff_zero_eq_constantCoeff, map_mul, map_sub, map_one,
+          constantCoeff_X, sub_zero, one_mul, coeff_one, ↓reduceIte]; rfl
+        · rw [sub_mul]; simp only [one_mul, map_sub, coeff_mk, coeff_one, hn, ↓reduceIte];
+          rw [show n = n - 1 + 1 by exact (Nat.succ_pred hn).symm, coeff_succ_X_mul];
+          simp only [Pi.one_apply, coeff_mk, sub_self]]
+    exact one_pow (d + 1)
+
+theorem invOneSubPow_val_eq_mk_choose_add :
+    (invOneSubPow d).val = (mk fun n => Nat.choose (d + n) d : S⟦X⟧) := rfl
+
+theorem invOneSubPow_val_zero_eq_invUnitSub_one :
+    (invOneSubPow 0).val = invUnitsSub (1 : Sˣ) := by
+  rw [invOneSubPow, invUnitsSub]
+  simp only [zero_add, Nat.choose_zero_right, Nat.cast_one, one_pow, divp_one]
+
+theorem invOneSubPow_eq_inv_one_sub_pow :
+    invOneSubPow d = ⟨1 - X, invOfUnit (1 - X) 1, @mul_invOfUnit S _ (1 - X) 1 <| by
+    simp only [map_sub, map_one, constantCoeff_X, sub_zero, Units.val_one], by
+    rw [mul_comm]; exact @mul_invOfUnit S _ (1 - X) 1 <| by
+      simp only [map_sub, map_one, constantCoeff_X, sub_zero, Units.val_one]⟩⁻¹ ^ (d + 1) := by
+  rw [inv_pow]; exact (DivisionMonoid.inv_eq_of_mul _ (invOneSubPow d) <| by
+  rw [← Units.val_eq_one, Units.val_mul, Units.val_pow_eq_pow_val];
+  exact (invOneSubPow d).inv_val).symm
+
+theorem invOneSubPow_inv_eq_one_sub_pow :
+    (invOneSubPow d).inv = (1 - X : S⟦X⟧) ^ (d + 1) := rfl
+
+theorem invOneSubPow_val_mul_one_sub_pow_eq_one :
+    (invOneSubPow d).val * ((1 - X : S⟦X⟧) ^ (d + 1)) = 1 :=
+  (invOneSubPow d).val_inv
```

Comment:
Why do we want this theorem? We already have `(invOneSubPow d).val_inv` and that's a shorter name than your theorem name. Same question for the next few theorems. But maybe they're useful?